tests/control.rs

```diff
@@ -134,3 +134,98 @@ pub(crate) async fn basics() -> Result<()> {
 
     Ok(())
 }
+
+pub(crate) async fn auctions() -> Result<()> {
```

Comment:
Can't express enough how useful these tests are for understanding the code. This is 💯  for consumers of this project